### PR TITLE
DAOS-4047 control: update server config files

### PR DESCRIPTION
**Pull request by tanabarr**
- add multiple io servers to example config files
- remove dead change ownership code and config params
- remove unused code from server package
- code review fixes remaining from pr 1662

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: utils/config/examples/daos_server_psm2.yml [hunk 1]
```diff
@@ -92,3 +85,27 @@ servers:
   # bdev_list: [/tmp/daos-bdev]	# generate nvme.conf as follows:
               # [AIO]
               #   AIO /tmp/aiofile AIO1 4096
+-
```
Comment (kjacque) [inline utils/config/examples/daos_server_psm2.yml]: So is the assumption that everyone will want to be running 2 IO servers?

Comment (tanabarr) [reply utils/config/examples/daos_server_psm2.yml]: that is the intended configuration for a dual socket system which is likely to be the most common target platform. that's probably the best bet for an example configuration IMO

**Comment by tanabarr** [comment]
gatekeeper please use PR title and description as commit message when landing

File: src/control/server/ctl_harness.go [hunk 2]
```diff
@@ -85,7 +84,8 @@ func (hc *harnessClient) prepareRequest(req RemoteHarnessReq) (*mgmtpb.RanksReq,
 	return &mgmtpb.RanksReq{Ranks: req.Ranks, Timeout: req.Timeout, Force: req.Force}, nil
 }
 
-func (hc *harnessClient) processResponse(rpcResp *mgmtpb.RanksResp) (system.MemberResults, error) {
+// processRPC takes a RanksResp proto message and converts to system.MemberResults.
+func (hc *harnessClient) processRPC(rpcResp *mgmtpb.RanksResp) (system.MemberResults, error) {
```
Comment (mjmac) [inline src/control/server/ctl_harness.go]: I'm curious about the motivation behind the name changes here. The other names seemed clearer to me. The purpose of ```processRPC()``` is a bit vague without context (is it handling a RPC?), whereas ```processResponse()``` is pretty clear.

They're not exported methods so this doesn't bother me enough to block on.

File: utils/config/examples/daos_server_psm2.yml [hunk 3]
```diff
@@ -85,3 +82,27 @@ servers:
...
+  targets: 8                # count of storage targets per each server
+  first_core: 25            # offset of the first core for service xstreams
+  nr_xs_helpers: 0          # count of offload/helper xstreams per target
+  fabric_iface: ib0         # map to OFI_INTERFACE=ib0
```
Comment (mjmac) [inline utils/config/examples/daos_server_psm2.yml]: This should be ib1, right?

File: utils/config/examples/daos_server_sockets.yml [hunk 4]
```diff
@@ -86,3 +86,25 @@ servers:
   # bdev_list: [/tmp/daos-bdev]	# generate nvme.conf as follows:
               # [AIO]
               #   AIO /tmp/aiofile AIO1 4096
+-
+  targets: 8                # count of storage targets per each server
+  first_core: 25            # offset of the first core for service xstreams
+  nr_xs_helpers: 0          # count of offload/helper xstreams per target
+  fabric_iface: eth0        # map to OFI_INTERFACE=eth0
```
Comment (mjmac) [inline utils/config/examples/daos_server_sockets.yml]: eth1?

Comment (tanabarr) [reply utils/config/examples/daos_server_sockets.yml]: that's what I have it configured as, should it be ibX?

File: utils/config/examples/daos_server_psm2.yml (hunk 3 above)
Comment (tanabarr) [reply utils/config/examples/daos_server_psm2.yml]: yes and the port number needs to be changed

File: utils/config/examples/daos_server_sockets.yml (hunk 4 above)
Comment (mjmac) [reply utils/config/examples/daos_server_sockets.yml]: What I meant is that the first server configuration already has eth0, so I think this should either be eth1 or else the port needs to be different.

Comment (tanabarr) [reply utils/config/examples/daos_server_sockets.yml]: yes will change port, probably good having an example that uses the same NIC, even if it is not the most performant configuration

File: src/control/server/ctl_harness.go (hunk 2 above)
Comment (tanabarr) [reply src/control/server/ctl_harness.go]: thought it made more sense at the time , but I will change back

File: utils/config/examples/daos_server_sockets.yml (hunk 4 above)
Comment (tanabarr) [reply utils/config/examples/daos_server_sockets.yml]: wait but won't the fabric interface being the same fail your new config validation?

Comment (tanabarr) [reply utils/config/examples/daos_server_sockets.yml]: reverted to having different NICs in order not to encourage configurations that might affect performance

**Pull request merged**